diff --git a/src/ImNodeFlow.inl b/src/ImNodeFlow.inl
index b85e46f..2bef51b 100644
--- a/src/ImNodeFlow.inl
+++ b/src/ImNodeFlow.inl
@@ -223,7 +223,7 @@ namespace ImFlow
     Pin* BaseNode::inPin(const U& uid)
     {
         PinUID h = std::hash<U>{}(uid);
-        auto it = std::ranges::find_if(m_ins, [&h](std::shared_ptr<Pin>& p)
+        auto it = std::find_if(m_ins.begin(), m_ins.end(), [&h](std::shared_ptr<Pin>& p)
                             { return p->getUid() == h; });
         assert(it != m_ins.end() && "Pin UID not found!");
         return it->get();
@@ -238,7 +238,7 @@ namespace ImFlow
     Pin* BaseNode::outPin(const U& uid)
     {
         PinUID h = std::hash<U>{}(uid);
-        auto it = std::ranges::find_if(m_outs, [&h](std::shared_ptr<Pin>& p)
+        auto it = std::find_if(m_outs.begin(), m_outs.end(), [&h](std::shared_ptr<Pin>& p)
                             { return p->getUid() == h; });
         assert(it != m_outs.end() && "Pin UID not found!");
         return it->get();
